pkg/sqlstore/store_user.go

```diff
@@ -17,6 +19,49 @@ type UserRows struct {
 	Rows    [][]interface{} `json:"rows"`
 }
 
+// UserValue wraps data from the db that may be raw json or any other value.
+type UserValue struct {
+	jsonValue  json.RawMessage
+	otherValue interface{}
+}
+
+// Value returns the underlying value.
+func (u *UserValue) Value() interface{} {
+	if u.jsonValue != nil {
+		return u.jsonValue
+	}
+	return u.otherValue
+}
+
+// Scan implements Scan.
+func (u *UserValue) Scan(src interface{}) error {
+	u.jsonValue = nil
+	u.otherValue = nil
+	switch src := src.(type) {
+	case string:
+		trimmed := strings.TrimLeft(src, " ")
+		if (strings.HasPrefix(trimmed, "{") || strings.HasPrefix(trimmed, "[")) && json.Valid([]byte(src)) {
+			u.jsonValue = []byte(src)
+		} else {
+			u.otherValue = src
+		}
+	case []byte:
+		tmp := src
+		u.otherValue = tmp
+	default:
+		u.otherValue = src
+	}
+	return nil
+}
+
+// MarshalJSON implements MarshalJSON.
+func (u *UserValue) MarshalJSON() ([]byte, error) {
+	if u.jsonValue != nil {
+		return u.jsonValue, nil
+	}
+	return json.Marshal(u.otherValue)
+}
```

Comment:
When marshaling to json, there is nothing to do for `jsonValue` nice we already identified it as valid json.